src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadata.cpp

```diff
@@ -970,13 +1039,27 @@ ColumnMapperPtr IcebergMetadata::getColumnMapperForObject(ObjectInfoPtr object_i
     return persistent_components.schema_processor->getColumnMapperById(iceberg_object_info->underlying_format_read_schema_id);
 }
 
-ColumnMapperPtr IcebergMetadata::getColumnMapperForCurrentSchema() const
+ColumnMapperPtr IcebergMetadata::getColumnMapperForCurrentSchema(StorageSnapshotPtr storage_snapshot, ContextPtr context) const
 {
+    auto iceberg_table_state = extractIcebergSnapshotIdFromMetadataObject(storage_snapshot);
+    // This is a temporary cludge for cluster functions because now the state on replicas is not synchronized with the primary state on the coordinator
+    if (!iceberg_table_state)
+    {
+        auto configuration_ptr = configuration.lock();
+        if (!configuration_ptr)
```

Comment:
In which case can we get `!configuration_ptr`? Does this mean this method was executes after query finished?

Reply:
This is really strange because configuration was always non-nullptr (and there were no such checks before), so it's important to save such invariant